crates/router/src/core/payments/routing.rs

```diff
@@ -1143,78 +1248,77 @@ async fn perform_session_routing_for_pm_type(
-//         chosen_connectors,
-//         session_pm_input.backend_input.clone(),
-//         None,
-//         session_pm_input.profile_id,
-//         transaction_type,
-//     )
-//     .await?;
-
-//     if final_selection.is_empty() {
-//         let fallback = profile_wrapper
-//             .get_default_fallback_list_of_connector_under_profile()
-//             .change_context(errors::RoutingError::FallbackConfigFetchFailed)?;
-
-//         final_selection = perform_cgraph_filtering(
-//             &session_pm_input.state.clone(),
-//             session_pm_input.key_store,
-//             fallback,
-//             session_pm_input.backend_input.clone(),
-//             None,
-//             session_pm_input.profile_id,
-//             transaction_type,
-//         )
-//         .await?;
-//     }
-
-//     if final_selection.is_empty() {
-//         Ok(None)
-//     } else {
-//         Ok(Some(final_selection))
-//     }
-// }
+#[cfg(feature = "v2")]
+async fn perform_session_routing_for_pm_type(
+    session_pm_input: &SessionRoutingPmTypeInput<'_>,
+    transaction_type: &api_enums::TransactionType,
+    business_profile: &domain::Profile,
+) -> RoutingResult<Option<Vec<api_models::routing::RoutableConnectorChoice>>> {
+    let merchant_id = &session_pm_input.key_store.merchant_id;
+
+    let MerchantAccountRoutingAlgorithm::V1(algorithm_id) = session_pm_input.routing_algorithm;
+
+    let profile_wrapper = admin::ProfileWrapper::new(business_profile.clone());
+    let chosen_connectors = if let Some(ref algorithm_id) = algorithm_id {
+        let cached_algorithm = ensure_algorithm_cached_v1(
+            &session_pm_input.state.clone(),
+            merchant_id,
+            algorithm_id,
+            session_pm_input.profile_id,
+            transaction_type,
+        )
+        .await?;
+
+        match cached_algorithm.as_ref() {
+            CachedAlgorithm::Single(conn) => vec![(**conn).clone()],
+            CachedAlgorithm::Priority(plist) => plist.clone(),
+            CachedAlgorithm::VolumeSplit(splits) => perform_volume_split(splits.to_vec())
+                .change_context(errors::RoutingError::ConnectorSelectionFailed)?,
+            CachedAlgorithm::Advanced(interpreter) => execute_dsl_and_get_connector_v1(
+                session_pm_input.backend_input.clone(),
+                interpreter,
+            )?,
+        }
+    } else {
+        profile_wrapper
+            .get_default_fallback_list_of_connector_under_profile()
+            .change_context(errors::RoutingError::FallbackConfigFetchFailed)?
+    };
+
+    let mut final_selection = perform_cgraph_filtering(
+        &session_pm_input.state.clone(),
+        session_pm_input.key_store,
+        chosen_connectors,
+        session_pm_input.backend_input.clone(),
+        None,
+        session_pm_input.profile_id,
+        transaction_type,
+    )
+    .await?;
+
+    if final_selection.is_empty() {
+        let fallback = profile_wrapper
```

Comment:
`chosen_connectors` itself derived from `get_default_fallback_list_of_connector_under_profile` if algorithm_id not present, how the below cases is different from that?